modules/dbparser/dbparser-grammar.ym

```diff
@@ -79,11 +79,15 @@ parser_db_opts
 /* NOTE: we don't support parser_opt as we don't want the user to specify a template */
 parser_db_opt
         : KW_FILE '(' string ')'                		{ log_db_parser_set_db_file(((LogDBParser *) last_parser), $3); free($3); }
-	| KW_INJECT_MODE '(' parser_db_inject_mode ')'		{ log_db_parser_set_inject_mode(((LogDBParser *) last_parser), $3); free($3); }
-	| parser_opt
+	| stateful_parser_opt
         ;
 
-parser_db_inject_mode
+stateful_parser_opt
+	: KW_INJECT_MODE '(' stateful_parser_inject_mode ')'	{ stateful_parser_set_inject_mode(((StatefulParser *) last_parser), $3); free($3); }
+	| parser_opt
+	;
+
+stateful_parser_inject_mode
 	: string				{ $$ = $1; }
```

Comment:
the reason KW_INTERNAL had an explicit rule was that it is a reserved word and the string rule wouldn't match it.

but indeed it is much better to report these issues during parsing, so that's what 33af212 does.

I wouldn't want to use explicit rules for individual values, those would generate a lot of keywords,
which might then collide with user identifiers.